english/tests/tests.py

```diff
@@ -5,121 +5,292 @@
-                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"},
+                {"pk": 4, "first_name": "Damian", "last_name": "Ward"},
+                {"pk": 5, "first_name": "Jonah", "last_name": "Matthews"},
+                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"}
             ],
-            "teacher": {"pk": 5, "first_name": "Damian", "last_name": "Ward"},
+            "teacher": {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
+            "days": ["Monday", "Wednesday", "Friday"],
+            "start_time": "18:00:00",
+            "end_time": "19:30:00"
         }
         url = reverse("classes-list")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 3)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_get_classes_for_current_student(self):
+    def test_get_classes_success(self):
+        self.client.force_login(user=self.student)
         expected_data = {
             "pk": 1,
             "name": "FirstClass",
             "students": [
-                {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
                 {"pk": 3, "first_name": "Ayden", "last_name": "Henris"},
-                {"pk": 4, "first_name": "Brody", "last_name": "Nelson"},
-                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"},
+                {"pk": 4, "first_name": "Damian", "last_name": "Ward"},
+                {"pk": 5, "first_name": "Jonah", "last_name": "Matthews"},
+                {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"}
             ],
-            "teacher": {"pk": 5, "first_name": "Damian", "last_name": "Ward"},
+            "teacher": {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
+            "days": ["Monday", "Wednesday", "Friday"],
+            "start_time": "18:00:00",
+            "end_time": "19:30:00"
         }
-        url = reverse("student-classes")
+        url = reverse("student-classes-list")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 2)
+        self.assertEqual(len(response.data), 3)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_get_classes_for_current_teacher(self):
+    def test_get_classes_failed(self):
+        self.client.force_login(user=self.teacher)
+        url = reverse("student-classes-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_leave_class(self):
+        self.client.force_login(user=self.student)
+        url = "http://0.0.0.0:8000/english/student/classes/1/join/"
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
After, please check that this user is not in the student list.
You need to check all changes which make your URL.
You need to get class by pk from DB and check that this user is not in the student list. (Before using this URL you need to be sure that this user in the student list for this class)